indy_node/server/request_handlers/config_req_handlers/auth_rule/abstract_auth_rule_handler.py

```diff
@@ -18,6 +20,8 @@ def __init__(self, database_manager: DatabaseManager,
         super().__init__(database_manager, txn_type, CONFIG_LEDGER_ID)
         self.write_req_validator = write_req_validator
         self.constraint_serializer = ConstraintsSerializer(domain_state_serializer)
+        self.config = getConfig()
+        self._update_state_by_versions = self._get_update_state_by_versions()
```

Comment:
Although this version processing can be fine for the fix, I think it would be great to eventually do it in a cleaner way using strategy pattern more explicitly:
1) Create AuthRule Handler classes for the version 1.9.1 (the only thing that they do differently is `update_state` method).
2) Add a possibility to register req handlers for the given state
3) Register AuthRule1.9.1
4) Apply a strategy in WriteRegManager's `restore_state`:
- get a version for the txn
- if there is a specific req handler for this version registered - call it
- otherwise call a common (current) handler
5) In future: extend the versioning pattern to other or all WriteRegManager's methods